File: app/src/main/java/com/example/servicebookingandroid/Request/RequestDetailActivity.java
package com.example.servicebookingandroid.Request;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.servicebookingandroid.Auth.AuthBaseActivity;
import com.example.servicebookingandroid.Model.RequestDto;
import com.example.servicebookingandroid.Model.UserDto;
import com.example.servicebookingandroid.R;
import com.example.servicebookingandroid.Request.Comment.CommentFormActivity;
import com.example.servicebookingandroid.Request.Comment.CommentsActivity;

public class RequestDetailActivity extends RequestBaseActivity {

    private int index = -1;
    TextView tv_title, tv_name, tv_date, tv_info, tv_location, tv_phone, tv_language, tv_comments;
    Button bt_edit, bt_comment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_request_detail);
    }

    public void initView() {
        tv_title = findViewById(R.id.tv_title);
        tv_name = findViewById(R.id.tv_name);
        tv_date = findViewById(R.id.tv_date);
        tv_info = findViewById(R.id.tv_info);
        tv_location = findViewById(R.id.tv_location);
        tv_phone = findViewById(R.id.tv_phone);
        tv_language = findViewById(R.id.tv_language);
        tv_comments = findViewById(R.id.tv_comments);

        bt_edit = findViewById(R.id.bt_edit);
        bt_comment = findViewById(R.id.bt_comment);

        if (AuthBaseActivity.user.getRole().equals("Service")) {
            bt_comment.setText("Post Comment");
            bt_edit.setVisibility(View.INVISIBLE);
        }
    }

    public void initValue() {
        Intent intent = getIntent();
        Integer i = intent.getIntExtra("index", 0);
        if (i != null) index = i;

        RequestDto requestDto = requestDtoList.get(index);

        tv_title.setText(requestDto.getServicetype());
        String date = "Post on: "+requestDto.getCreate_At();
        if (requestDto.getUpdate_At() != null) date += "\nUpdated on: " + requestDto.getUpdate_At();
        tv_date.setText(date);
        UserDto userDto = requestDto.getUserDto();
        tv_name.setText(userDto.getFirstname() + " " + userDto.getLastname());
        String location = userDto.getStreetname() + ", " + userDto.getCity() + ", " + userDto.getState() + ", " + userDto.getZipcode();
        tv_location.setText("Location: " + location);
        tv_phone.setText("Phone Number: " + userDto.getPhone());
        tv_language.setText("Language: " + userDto.getLanguage());
        tv_info.setText(requestDto.getInfo());
    }

    public void onEditRequest(View view) {
        Intent i = new Intent(this, RequestFormActivity.class);
        i.putExtra("index", index);
        startActivity(i);
    }

    public void onComments(View view) {
        String role = AuthBaseActivity.user.getRole();
        if (role.equals("Customer")) {
            Intent i = new Intent(this, CommentsActivity.class);
            i.putExtra("requestId", String.valueOf(requestDtoList.get(index).getRequestId()));
            startActivity(i);
        } else if (role.equals("Service")){
            Intent i = new Intent(this, CommentFormActivity.class);
            i.putExtra("requestId", String.valueOf(requestDtoList.get(index).getRequestId()));
            startActivity(i);
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        initView();
        initValue();
    }
}

File: app/src/main/java/com/example/servicebookingandroid/Request/RequestDetailActivity.kt
package com.example.servicebookingandroid.Request

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.example.servicebookingandroid.Auth.AuthBaseActivity
import com.example.servicebookingandroid.Model.RequestDto
import com.example.servicebookingandroid.R
import com.example.servicebookingandroid.Request.Comment.CommentFormActivity
import com.example.servicebookingandroid.Request.Comment.CommentsActivity
import kotlinx.android.synthetic.main.activity_request_detail.*

class RequestDetailActivity : RequestBaseActivity() {

    var index: Int = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_request_detail)
    }

    fun initView() {
        if (AuthBaseActivity.user.role == "Service") {
            bt_comment.setText("Post Comment")
            bt_edit.setVisibility(View.INVISIBLE)
        }
    }

    fun initValue() {
        var intent: Intent = getIntent()
        var i: Int = intent.getIntExtra("index", -1)
        if (i != -1) index = i

        var requestDto: RequestDto = requestDtoList[index]

        tv_title.text = requestDto.servicetype
        var date = "Post on: " + requestDto.create_At
        if (requestDto.update_At != null) date += "\nUpdated on: " + requestDto.update_At
        tv_date.text = date
        val userDto = requestDto.userDto
        tv_name.text = userDto.firstname + " " + userDto.lastname
        val location = userDto.streetname + ", " + userDto.city + ", " + userDto.state + ", " + userDto.zipcode
        tv_location.text = "Location: $location"
        tv_phone.text = "Phone Number: " + userDto.phone
        tv_language.text = "Language: " + userDto.language
        tv_info.text = requestDto.info
    }

    fun onEditRequest(view: View) {
        val i = Intent(this, RequestFormActivity::class.java)
        i.putExtra("index", index)
        startActivity(i)
    }

    fun onComments(view: View) {
        val role = AuthBaseActivity.user.role
        if (role == "Customer") {
            val i = Intent(this, CommentsActivity::class.java)
            i.putExtra("requestId", requestDtoList[index].requestId.toString())
            startActivity(i)
        } else if (role == "Service") {
            val i = Intent(this, CommentFormActivity::class.java)
            i.putExtra("requestId", requestDtoList[index].requestId.toString())
            startActivity(i)
        }
    }

    override fun onStart() {
        super.onStart()
        initView()
        initValue()
    }
}
